protobuf/lib/src/protobuf/protobuf_enum.dart

```diff
@@ -37,9 +37,16 @@ class ProtobufEnum {
   /// Creates a new constant [ProtobufEnum] using [value] and [name].
   const ProtobufEnum(this.value, this.name);
 
-  /// Creates a Map for all of the [ProtobufEnum]s in [byIndex], mapping each
-  /// [ProtobufEnum]'s [value] to the [ProtobufEnum].
-  static Map<int, T> initByValue<T extends ProtobufEnum>(List<T> byIndex) {
+  static List<T?> initByValueList<T extends ProtobufEnum>(List<T> byIndex) {
+    if (byIndex.isEmpty) return [];
+    final byValue = List<T?>.filled(byIndex.last.value + 1, null);
+    for (final enumValue in byIndex) {
+      byValue[enumValue.value] = enumValue;
+    }
+    return byValue;
+  }
+
+  static Map<int, T> initByValueMap<T extends ProtobufEnum>(List<T> byIndex) {
     final byValue = <int, T>{};
```

Comment:
This could also use `List<T>` where we do binary search in. It may be much more compact in memory, quite possibly also faster.

The binary searching code could also live here as a static
```
static T? binarySearchFor<T extendfs ProtobufEnum>(List<T> byValue) {
  ...
}
```

Reply:
This is done in #985 and #987, but the results are not that much better.